operator/controllers/loki/lokistack_dashboards_controller.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	"github.com/go-logr/logr"
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/manifests/openshift"
+	"k8s.io/apimachinery/pkg/labels"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var deletesOnlyPred = builder.WithPredicates(predicate.Funcs{
+	UpdateFunc:  func(e event.UpdateEvent) bool { return false },
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+})
+
+// LokiStackDasboardsReconciler cleans up all remaining
+// dashboards when all LokiStack objects are removed.
+type LokiStackDasboardsReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// Reconcile removes all LokiStack dashboard ConfigMap objects on OpenShift clusters when
+// the last LokiStack custom resource is deleted.
+func (r *LokiStackDasboardsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var stacks lokiv1.LokiStackList
+	err := r.List(ctx, &stacks, client.MatchingLabelsSelector{Selector: labels.Everything()})
+	if err != nil {
+		return ctrl.Result{}, kverrors.Wrap(err, "failed to list any lokistack instances", "req")
+	}
+
+	if len(stacks.Items) > 0 {
+		return ctrl.Result{}, nil
+	}
```

Comment:
nit: Do we really need this check 🤔 feel like we could just try to delete thing and if they are already not there just exit nicely

Reply:
We want to delete only once all LokiStack resources are deleted as per these Dashboards shared by all LokiStack instances in the cluster. For example if someone deletes the LokiStack instance in `openshift-logging` but keeps the instance in `openshit-network-observability` we do not want to delete the dashboards.

Reply:
Yes... obviously! I completely forgot that thanks for clarifying!